src/plugins/splitOperation.ts

```diff
@@ -0,0 +1,98 @@
+import { Host, Session } from "@azure-tools/autorest-extension-base";
+import { CodeModel, Request, Operation, Parameter } from "@azure-tools/codemodel";
+import { isNullOrUndefined } from "util";
+import { Helper } from "../helper";
+import { CliConst, CliCommonSchema } from "../schema";
+import { NodeHelper } from "../nodeHelper";
+import { Modifier } from "./modifier/modifier";
+import { CopyHelper } from "../copyHelper";
+
+export class SplitOperation{
+
+    constructor(protected session: Session<CodeModel>){
+    }
+
+    public async process() {
+
+        await this.modifier();
+
+        for (const group of this.session.model.operationGroups) {
```

Comment:
>for (const group of this.session.model.operationGroups [](start = 8, length = 54)

I remember there is a utility method to go through code model, please check whether that can be reused.

Reply:
I find `enumerateCodeModel` from `helper.ts`. If I want to get cli property from node, I think this method is suitable. But here I do need cli properties. I think access code model directly is more clear.

Reply:
i remember there is a flag in that method that can control which part to iterate, is that what you want?

---
In reply to: [433050349](https://github.com/Azure/autorest.clicommon/pull/65#discussion_r433050349) [](ancestors = 433050349)

Reply:
maybe not suitable considering you need to something related in both group and operation level

---
In reply to: [433596320](https://github.com/Azure/autorest.clicommon/pull/65#discussion_r433596320) [](ancestors = 433596320,433050349)